docker/Dockerfiles/Dockerfile-dev

```diff
@@ -3,7 +3,7 @@
 # Install Rally from source inside a virtualenv
 ################################################################################
 
-FROM python:3.8.2-slim as builder
+FROM python:3.8.12-slim-bullseye as builder
```

Comment:
python:3.8-slim avoid having to specify a patch version and already defaults to bullseye:

```
docker run python:3.8-slim cat /etc/debian_version   
11.0
```

Any reason not to rely on the defaults here?

Reply:
local caching, in case I run docker tests in an environment where I already have an image tagged `3.8-slim` will prevent updating the patch version.  It's good practice to be explicit with things like this

Reply:
I agree about being explicit being the safer thing to do, thanks @DJRickyB

Reply:
Definitely best practice (I've used hadolint in the past to enforce this), was wondering if it made sense here. It does, thanks.